docs/src/rules/no-restricted-imports.md

```diff
@@ -282,6 +282,60 @@ import { AllowedObject } from "foo";
 
 :::
 
+#### allowTypeImports (TypeScript only)
+
+Whether to allow [Type-Only Imports](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export) for a path. Default: `false`.
```

Comment:
[Docs] I'm only now realizing this, but I think there's an ambiguity in the existing typescript-eslint docs: does "type-only imports" also include "type-only exports"? Intuitively I don't think so, but all the docs and implementations say so. IIRC it's because an `export ... from` by nature includes an `import`... but that's not something every user would know.

Proposal: can we mention this in the docs here, to fix this long-standing potential confusion?

Maybe:

> This includes type-only `export`s, as they are equivalent to re-exporting an `import`.

Reply:
Updated in https://github.com/eslint/eslint/pull/19562/commits/921496ade00f69b09dcc52130391da397350905d